### PR TITLE
Adds private check to MessageList#bulkDelete and adds IUser#getPermissionsForGuild

### PR DESCRIPTION
**Pull request by nija123098**
### Prerequisites
* [x] I have read and understood the [contribution guidelines](CONTRIBUTING.md)
* [x] This pull request follows the code style of the project
* [x] I have tested this feature thoroughly
  * http://i.imgur.com/lbrqN8j.png
  * http://i.imgur.com/NgW54mK.png

### Changes Proposed in this Pull Request
* Fixes issue where MessageList#bulkDelete would not check if the channel is private
* Adds IUser#getPermissionsForGuild and User implementation to do the obvious

### CODE REVIEW

**Comment by nija123098** [comment]
Er, I might have re-based incorrectly, thus the 15 commits, however the changes are as intended.

File: src/main/java/sx/blah/discord/handle/obj/IUser.java [hunk 1]
```diff
@@ -106,6 +107,14 @@
 	List<IRole> getRolesForGuild(IGuild guild);
 
 	/**
+	 * Gets the permissions the user has on the guild.
+	 *
+	 * @param guild The guild to check the permissions for.
+	 * @return The permissions.
+     */
```
Comment (austinv11) [inline src/main/java/sx/blah/discord/handle/obj/IUser.java]: This indentation irks me

File: src/main/java/sx/blah/discord/handle/impl/obj/User.java [hunk 2]
```diff
@@ -276,6 +276,13 @@ public void setDiscriminator(String discriminator) {
 	}
 
 	@Override
+	public EnumSet<Permissions> getPermissionsForGuild(IGuild guild){
```
Comment (austinv11) [inline src/main/java/sx/blah/discord/handle/impl/obj/User.java]: You should update Channel#getModifiedPermissions to use this method for consistency.

File: src/main/java/sx/blah/discord/handle/obj/IUser.java (hunk 1 above)
Comment (nija123098) [inline src/main/java/sx/blah/discord/handle/obj/IUser.java]: Intellij didn't show this.  I had to go into note pad.

**Comment by nija123098** [comment]
Previous commit at request of @GrandPanda.

**Pull request merged**